### DISCUSSION TITLE
How to override the export url

### DISCUSSION THREAD

**Discussion by PrasadChinwal** [filamentphp/filament#11494] [Help] [answered]
### Package

Actions

### Package Version

v3.0.0

### How can we help you?

I have filament running inside a directory on server. Example my url is `https://xyz.com/app` and so far filament works perfectly. 

I am struggling to override the url of the download button on export notification as seen below:
<img width="450" alt="Screenshot 2024-02-19 at 10 18 26 PM" src="https://github.com/filamentphp/filament/assets/7385673/662945e7-baca-4e1b-920e-dbb49313ff9b">

The download csv or download xlsx buttons give the below url:
`https://xyz.com/filament/exports/12/download?format=csv` 

and I want to override it to 

`https://xyz.com/app/filament/exports/12/download?format=csv`.

**Comment by leandrocfe** [comment, marked as answer]
I guess you can override this route in the `routes/web.php` file

```php
use Filament\Actions\Exports\Http\Controllers\DownloadExport;

Route::get('/app/filament/exports/{export}/download', DownloadExport::class)
    ->name('filament.exports.download')
    ->middleware(['web', 'auth']);
```

**Reply by PrasadChinwal** [reply]
Yes! Thanks a lot @leandrocfe this worked for me 👍

**Reply by Mohammed-800** [reply]
@leandrocfe 

  Unable to prepare route [app/filament/exports/{export}/download] for serialization. Another route has already been assigned name [filament.exports.download].

**Reply by binaryfire** [reply]
@Mohammed-800 Hi. Did you find a solution for this?

**Reply by PrasadChinwal** [reply]
@Mohammed-800, @binaryfire check your routes using `php artisan route:list` and see if your export route is different than `/app/filament/exports/{export}/download`.

**Comment by tjarrett** [comment]
I'm hitting this issue too and the problem seems to be in the enum `ExportFormat` in this chunk of code:

```
    public function getDownloadNotificationAction(Export $export): NotificationAction
    {
        return NotificationAction::make("download_{$this->value}")
            ->label(__("filament-actions::export.notifications.completed.actions.download_{$this->value}.label"))
            ->url(route('filament.exports.download', ['export' => $export, 'format' => $this], absolute: false), shouldOpenInNewTab: true)
            ->markAsRead();
    }
```

Specifically the `absolute: false` is causing this issue for me. If I remove it the download link works as expected. Looking through the rest of Filament code and interface, most links seem to be absolute by default. Is there a reason these notification links should NOT be absolute? I've tried to think-up what the reason might be and I cannot come up with anything.

**Reply by binaryfire** [reply]
This has been solved: https://github.com/filamentphp/filament/pull/15860

Should be merged pretty soon.

**Comment by pranab-acharya** [comment]
Yes, it is working now. I was facing an issue with the stancl/tenancy ("tenancy-for-laravel") package when handling exports. To resolve it, I added the export route to the tenancy middleware and extended DownloadExport with a custom class DownloadExportCustom.

Here’s the relevant part of the code:

public function __invoke(Request $request, Export $export): StreamedResponse
    {
        if (filled(Gate::getPolicyFor($export::class))) {
            authorize('view', $export);
        } else {
            abort_unless($export->user()->is(Filament::auth()->user()), 403);
        }

        $format = $this->resolveFormatFromRequest($request);

        abort_unless($format !== null, 404);

        return $format->getDownloader()($export);
    }

After making these changes, everything started working as expected.
